src/main/java/studio/magemonkey/fusion/api/EventManager.java

```diff
@@ -0,0 +1,256 @@
+package studio.magemonkey.fusion.api;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import studio.magemonkey.codex.CodexEngine;
+import studio.magemonkey.codex.api.DelayedCommand;
+import studio.magemonkey.codex.util.messages.MessageData;
+import studio.magemonkey.codex.util.messages.MessageUtil;
+import studio.magemonkey.fusion.Fusion;
+import studio.magemonkey.fusion.api.events.*;
+import studio.magemonkey.fusion.cfg.sql.SQLManager;
+import studio.magemonkey.fusion.data.player.FusionPlayer;
+import studio.magemonkey.fusion.data.professions.Profession;
+import studio.magemonkey.fusion.data.queue.CraftingQueue;
+import studio.magemonkey.fusion.data.queue.QueueItem;
+import studio.magemonkey.fusion.data.recipes.CraftingTable;
+import studio.magemonkey.fusion.util.ExperienceManager;
+import studio.magemonkey.fusion.util.PlayerUtil;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class EventManager {
+
+    /**
+     * Call the ProfessionJoinEvent.
+     * @param professionName The name of the profession.
+     * @param player The player that wants to join the profession.
+     * @param moneyCost The money cost to join the profession.
+     * @param expCost The experience cost to join the profession.
+     */
+    public void callProfessionJoinEvent(String professionName, Player player, double moneyCost, int expCost) {
```

Comment:
These methods are all a little weird to me. The name suggest that all that is happening is that the event is being fired off for hooked plugins to intercept, not that it's actually doing the update logic itself. Additionally, this class being called `EventManager` is a little strange. While, yes, it is calling events, it is doing more than that because it is performing the subsequent logic based on the results of the events. And it has a variety of focuses as well.

I generally try to keep similar things together for sake of readability and organization. Profession-related methods and events could be part of a `ProfessionService` class that handles doing the professing/experience related logic as well as calling those events. A `QueueService` might manage queue-related logic/events. This way, if we're looking to resolve an issue with queues somewhere, we can just go to the QueueService and not have to dig through profession-related code.

Reply:
This makes totally sense, I have seperated those into services and appended thom on a class `EventServices` which contains `ProfessionService` and `QueueService`